### PR TITLE
Move and reference Agents -> Workers upgrade guide

### PR DESCRIPTION
**Pull request by billpalombi**
The agents Agents -> Workers upgrade guide belongs in the guide section. We can link to it from wherever we'd like. I've added links to the top of the agents, infrastructure, and block based deployment pages.

### CODE REVIEW

File: docs/concepts/deployments-block-based.md [hunk 1]
```diff
@@ -17,6 +17,9 @@ search:
 
 # Block Based Deployments
 
+!!! note "Workers are reccomended"
+    This page is about the block-based deployment model. The [Work Pools and Workers](/concepts/work-pools/) based [deployment model](/concepts/deployments/) simplifies the specification of flow's infrastructure and runtime environment. If you have existing agents, you can [upgrade from agents to workers](/guides/upgrade-guide-agents-to-workers/) to significantly enhance the experience of deploying flows.
```
Comment (discdiver) [inline docs/concepts/deployments-block-based.md:21]:
```suggestion
    This page is about the block-based deployment model. The [Work Pools and Workers](/concepts/work-pools/) based [deployment model](/concepts/deployments/) simplifies the specification of a flow's infrastructure and runtime environment. If you have existing agents, you can [upgrade from agents to workers](/guides/upgrade-guide-agents-to-workers/) to significantly enhance the experience of deploying flows.
```

Comment (discdiver) [reply docs/concepts/deployments-block-based.md:21]: I think it sounds more natural with the article, but wouldn't die on the hill.

File: docs/concepts/infrastructure.md [hunk 2]
```diff
@@ -16,6 +16,9 @@ search:
 
 # Infrastructure
 
+!!! note "Workers are reccomended"
```
Comment (serinamarie) [inline docs/concepts/infrastructure.md:19]: little typo

File: docs/concepts/agents.md [hunk 3]
```diff
@@ -7,7 +7,11 @@ search:
   boost: 2
 ---
 
-# Agents 
+# Agents
+
+!!! note "Workers are reccomended"
```
Comment (serinamarie) [inline docs/concepts/agents.md:12]: little typo

**Pull request merged**